applications/IgaApplication/custom_strategies/custom_strategies/eigensolver_nitsche_stabilization_strategy.hpp

```diff
@@ -0,0 +1,665 @@
+    }
+
+    void SetReformDofSetAtEachStepFlag(bool flag)
+    {
+        this->pGetBuilderAndSolver()->SetReshapeMatrixFlag(flag);
+    }
+
+    bool GetReformDofSetAtEachStepFlag() const
+    {
+        return this->pGetBuilderAndSolver()->GetReshapeMatrixFlag();
+    }
+
+    /// Set verbosity level of the solving strategy.
+    /**
+     * - 0 -> mute... no echo at all
+     * - 1 -> print time and basic information
+     * - 2 -> print linear solver data
+     * - 3 -> print debug information
+     */
+    void SetEchoLevel(int Level) override
+    {
+        BaseType::SetEchoLevel(Level);
+        this->pGetBuilderAndSolver()->SetEchoLevel(Level);
+    }
+
+    /**
+     * Initialization to be performed once before using the strategy.
+     */
+    void Initialize() override
+    {
+        KRATOS_TRY
+
+        ModelPart& rModelPart = BaseType::GetModelPart();
+        const int rank = rModelPart.GetCommunicator().MyPID();
+
+        KRATOS_INFO_IF("EigensolverNitscheStabilizationStrategy", BaseType::GetEchoLevel() > 2 && rank == 0)
+            <<  "Entering Initialize" << std::endl;
+
+        if (mInitializeWasPerformed == false)
+        {
+            SchemePointerType& pScheme = this->pGetScheme();
+
+            if (pScheme->SchemeIsInitialized() == false)
+                pScheme->Initialize(rModelPart);
+
+            if (pScheme->ElementsAreInitialized() == false)
+                pScheme->InitializeElements(rModelPart);
+
+            if (pScheme->ConditionsAreInitialized() == false)
+                pScheme->InitializeConditions(rModelPart);
+        }
+
+        KRATOS_INFO_IF("EigensolverNitscheStabilizationStrategy", BaseType::GetEchoLevel() > 2 && rank == 0)
+            <<  "Exiting Initialize" << std::endl;
+
+        KRATOS_CATCH("")
+    }
+
+    /**
+     * Clears the internal storage
+     */
+    void Clear() override
+    {
+        KRATOS_TRY
+
+        // if the preconditioner is saved between solves, it should be cleared here
+        BuilderAndSolverPointerType& pBuilderAndSolver = this->pGetBuilderAndSolver();
+        pBuilderAndSolver->GetLinearSystemSolver()->Clear();
+
+        if (this->pGetStiffnessMatrix() != nullptr)
+            this->pGetStiffnessMatrix() = nullptr;
+
+        if (this->pGetStabilizationMatrix() != nullptr)
+            this->pGetStabilizationMatrix() = nullptr;
+
+        // Re-setting internal flag to ensure that the dof sets are recalculated
+        pBuilderAndSolver->SetDofSetIsInitializedFlag(false);
+
+        pBuilderAndSolver->Clear();
+
+        this->pGetScheme()->Clear();
+
+        mInitializeWasPerformed = false;
+
+        KRATOS_CATCH("")
+    }
+
+    /**
+     * Performs all the required operations that should be done (for each step)
+     * before solving the solution step.
+     * A member variable should be used as a flag to make sure this function is called only once per step.
+     */
+    void InitializeSolutionStep() override
```

Comment:
Please highleight specifically what you do here. Soon you will be happy about your comments.